frontend/packages/local-storage-operator-plugin/src/components/local-volume-set/create-local-volume-set.tsx

```diff
@@ -1,94 +1,88 @@
-  const [volumeMode, setVolumeMode] = React.useState(volumeModeDropdownItems.Block);
-  const [maxVolumeLimit, setMaxVolumeLimit] = React.useState('');
-  const [nodeNames, setNodeNames] = React.useState<string[]>([]);
+  const [state, dispatch] = React.useReducer(reducer, initialState);
+  const [nodeData, nodeLoaded, nodeLoadError] = useK8sWatchResource<NodeKind[]>(nodeResource);
 
-  const { ns, appName } = match.params;
+  const { appName, ns } = match.params;
   const modelName = LocalVolumeSetModel.label;
 
-  const toggleShowNodesList = () => {
-    setShowNodesList(!showNodesList);
-  };
+  React.useEffect(() => {
+    if ((nodeLoadError || nodeData.length === 0) && nodeLoaded) {
+      dispatch({ type: 'setAllNodeNames', value: [] });
+    } else if (nodeLoaded) {
+      const allNodeNames = nodeData.filter((node) => !hasTaints(node)).map((node) => getName(node));
+      dispatch({ type: 'setAllNodeNames', value: allNodeNames });
+    }
+  }, [nodeData, nodeLoaded, nodeLoadError]);
 
   const onSubmit = (event: React.FormEvent<EventTarget>) => {
     event.preventDefault();
 
     const requestData: LocalVolumeSetKind = {
-      apiVersion: LocalVolumeSetModel.apiVersion,
+      apiVersion: apiVersionForModel(LocalVolumeSetModel),
       kind: LocalVolumeSetModel.kind,
-      metadata: { name: volumeSetName },
+      metadata: { name: state.volumeSetName, namespace: LSO_NAMESPACE },
       spec: {
-        storageClassName,
-        volumeMode,
+        storageClassName: state.storageClassName || state.volumeSetName,
+        volumeMode: state.diskMode,
         deviceInclusionSpec: {
-          // Only Raw disk supported for 4.5
-          deviceTypes: [DeviceType.RawDisk],
-          deviceMechanicalProperty: [DeviceMechanicalProperty[volumeType]],
+          // Only Raw disk supported for 4.6
+          deviceTypes: [DiskType.RawDisk],
+          deviceMechanicalProperty:
+            state.diskType === 'HDD'
+              ? [DiskMechanicalProperty[state.diskType]]
+              : [DiskMechanicalProperty.SSD],
+        },
+        nodeSelector: {
+          nodeSelectorTerms: [
+            {
+              matchExpressions: [
+                { key: 'kubernetes.io/hostname', operator: 'In', values: state.nodeNames },
+              ],
+            },
+          ],
         },
       },
     };
 
-    if (showNodesList) {
-      requestData.spec.nodeSelector = {
-        nodeSelectorTerms: [
-          {
-            matchExpressions: [
-              { key: 'kubernetes.io/hostname', operator: 'In', values: nodeNames },
-            ],
-          },
-        ],
-      };
-    }
-    if (maxVolumeLimit) requestData.spec.maxDeviceCount = +maxVolumeLimit;
+    if (state.maxDiskLimit) requestData.spec.maxDeviceCount = +state.maxDiskLimit;
+    if (state.minDiskSize)
+      requestData.spec.deviceInclusionSpec.minSize = state.minDiskSize.toString();
+    if (state.maxDiskSize && state.maxDiskSize !== 'All')
```

Comment:
this value should be a constant.
```suggestion
    if (state.maxDiskSize && state.maxDiskSize !== MAX_DISK_SIZE)
```